webpack/scenes/ContentViews/Details/Filters/Add/CVFilterAddModal.js

```diff
@@ -0,0 +1,160 @@
+import React, { useState } from 'react';
+import useDeepCompareEffect from 'use-deep-compare-effect';
+import { Redirect } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { STATUS } from 'foremanReact/constants';
+import PropTypes from 'prop-types';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { Modal, ModalVariant, Form, FormGroup, TextInput, ActionGroup, Button, Radio, TextArea,
+  Split, SplitItem, Select, SelectOption } from '@patternfly/react-core';
+import { createContentViewFilter } from '../../ContentViewDetailActions';
+import {
+  selectCreateContentViewFilter, selectCreateContentViewFilterStatus,
+  selectCreateContentViewFilterError,
+} from '../../../Details/ContentViewDetailSelectors';
+import { FILTER_TYPES } from '../../../ContentViewsConstants';
+import ContentType from '../ContentType';
+
+const CVFilterAddModal = ({ cvId, show, setIsOpen }) => {
+  const selectOptions = Object.keys(FILTER_TYPES).map(item =>
+    <SelectOption key={item} value={item}><ContentType type={item} /></SelectOption>);
+  const [inclusion, setInclusion] = useState(true);
+  const [name, setName] = useState('');
+  const [description, setDescription] = useState('');
+  const [type, setType] = useState('rpm');
+  const [saving, setSaving] = useState(false);
+  const [typeSelectOpen, setTypeSelectOpen] = useState(false);
+  const dispatch = useDispatch();
+  const response = useSelector(state => selectCreateContentViewFilter(state));
+  const status = useSelector(state => selectCreateContentViewFilterStatus(state));
+  const error = useSelector(state => selectCreateContentViewFilterError(state));
+  const [redirect, setRedirect] = useState(false);
+
+  const onSave = () => {
+    setSaving(true);
+    dispatch(createContentViewFilter(
+      cvId,
+      {
+        name, description, inclusion, type,
+      },
+    ));
+  };
+
+  const onSelect = (event, selection) => {
+    setType(selection);
+    setTypeSelectOpen(false);
+  };
+
+  useDeepCompareEffect(() => {
+    const { id } = response || {};
```

Comment:
Will destructuring do the same thing?
```suggestion
    const { id = {} } = response;
```

Reply:
We have been following this pattern in several pages and I find it easy to understand. The double destructuring unless it provides performance benefits is more confusing than the alternative (to not so fluent in ES aka me)..Thoughts?

Reply:
okay, I can see that :)